### ISSUE TITLE
Groups and Tabs

### ISSUE THREAD

**Issue by ghost** [antoyo/titanium#8]
Placeholder for discussion on if/how to implement tabs and their state on the server, as well as grouping them (see: [min](https://github.com/minbrowser/min) "Tasks") by name.

**Comment by antoyo** [comment]
That would need to be added to the [`mg` crate](https://github.com/antoyo/mg).

**Comment by ghost** [comment]
For the state handling? I don't mean UI representation.

**Comment by antoyo** [comment]
So you don't want a UI for tabs, then?

If I understand correctly, I want something similar to vim buffers?

**Comment by ghost** [comment]
Yeah, vim buffers. Although I was also planning on opening an issue on `mg` for tabs.

**Comment by antoyo** [comment]
We'll probably use the [Stack widget](http://gtk-rs.org/docs/gtk/struct.Stack.html) for that.

I think we'll need a setting like `layout` that could be equal to `tab`, `stack` or `window`.
When in stack mode, instead of creating new `Mg` views, we'll just create a new webview and add it to the `Stack`.

For the tabs, I think we can delay that unless you want them?

For groups, I'm still not sure how it should work.
How do you want them to work?

**Comment by naturallymitchell** [comment]
Another way to add spatial organization, using this interface element and others, would be to use the tile/window manager itself.  This way, it's free (by convention/composition) and it solves the general issue of applications needing more navigable space, using good common tools.  Some more background interest:
* [Ditching browser tabs for i3 tabs][i3-tabs]
* [Why [Application-specific] Tabs Must Die [and What Should Replace Them]][app-tabs]

One way this could come together would be with [focus-oriented rules][tyrannical] in a [Rust Wayland Compositor][way-cooler] using way-cooler/way-cooler#394's way-cooler/way-cooler#48.  (aside: On top of that, we could also reorient screens on-the-fly in [cycles][cycle-focus], [overviews][revelation], and any other imaginable view.)

[app-tabs]:http://www.omgubuntu.co.uk/2012/04/why-tabs-must-die
[i3-tabs]:https://faq.i3wm.org/question/1578/ditching-browser-tabs-for-i3-tabs/index.html
[tyrannical]:https://github.com/Elv13/tyrannical
[way-cooler]:https://github.com/way-cooler/way-cooler
[cycle-focus]:https://github.com/blueyed/awesome-cyclefocus
[revelation]:https://github.com/guotsuan/awesome-revelation

**Comment by antoyo** [comment]
@mitchtbaum I actually use i3, that's why I don't want to implement that: if someone wants it, (s)he'll have to implement it.

**Comment by naturallymitchell** [comment]
@antoyo Could you please describe your set up a bit and any particular benefits and drawbacks it has for how you use Titanium?

**Comment by antoyo** [comment]
Well, I mostly use the tab layout in the window manager.
The drawback is that if I have many tabs, that can be confusing since they could be mixed with the other applications: there's a way to group tabs in i3 if this is an issue for you.
The benefit is that I can manipulate the tabs the way I want, from the keyboard.
